### PR TITLE
chore: move doctor to engine-server

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
# Pull Request Checklist

You can go to [dendron pull requests](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html) to see full details for items in this checklist.

### Quality Assurance
- [ ] add a [test](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#writing-tests) for the new feature
- [x] make sure all the existing [tests](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#running-all-tests) pass
- [x] do a spot check by running your feature with our [test workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace)
- [x] after you submit your pull request, check the output of our [integration test](https://github.com/dendronhq/dendron/actions) and make sure all tests pass
  - NOTE: if you running MacOS or Linux, pay special attention to the Windows output and vice versa if you are developing on Windows

#### Special Cases
- [ ] if your tests changes an existing snapshot, make sure that snapshots are [updated](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#updating-test-snapshots)
- [ ] if you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in [test-workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace). We use this to manually inspect new changes and for auto regression testing

### Docs
- [x] Make sure that the PR title follows our [commit style](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style)
- [ ] Please summarize the feature or impact in 1-2 lines in the PR description
- [ ] If your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

Example PR Description
```markdown
# feat: capitalize all foos

This changes capitalizes all occurrences of `foo` to `Foo`

Docs PR: <URL_TO_DOCS_PR>
```

### First Time PR
- [ ] sign the [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement) which will be prompted by our github bot after you submit the PR
- [ ] add your [discord](https://discord.gg/AE3NRw9) alias in the review so that we can give you the [horticulturalist](https://wiki.dendron.so/notes/7c00d606-7b75-4d28-b563-d75f33f8e0d7.html#horticulturalist) badge in our community

### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### CODE REVIEW

File: packages/engine-server/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "@dendronhq/common-all": "^0.69.1",
     "@dendronhq/common-server": "^0.69.1",
     "@dendronhq/remark-mermaid": "^0.2.0",
+    "@jcoreio/async-throttle": "^1.4.3",
```
Comment (jonathanyeung) [inline packages/engine-server/package.json:43]: Can you make a corresponding change to `package.json` of plugin-core to remove the cli dependency and make sure the build still works?

https://github.com/dendronhq/dendron/blob/1ae79c528d15b3a77a13362b1170dfe064913391/packages/plugin-core/package.json#L903

**Review by jonathanyeung — CHANGES_REQUESTED** [review]
One comment on removing the cli-dependency from plugin-core.  Otherwise looks good!

File: packages/engine-server/package.json (hunk 1 above)
Comment (kevinslin) [reply packages/engine-server/package.json:43]: yep, can confirm. go ahead and delete

**Comment by kevinslin** [comment]
i think we should be good to here as long as we rebase latest from master

**Review by kevinslin — APPROVED** [review]

**Review by jonathanyeung — APPROVED** [review]

**Pull request merged**